### DISCUSSION TITLE
German Translation causes Comment Submit Button to Overflow

### DISCUSSION THREAD

**Discussion by itssme** [requarks/wiki#3313] [Error / Bug Report]
**Describe the bug**
The translation for the "Submit Comment Button" in German is very long and causes the button to overflow and widen the page which enables the user to "scroll to the right" and see a wide empty space on devices with a small screen (e.g. some phones).

**To Reproduce**
Steps to reproduce the behaviour:
1. Enable German locals
2. View a wiki site where commenting is enabled on a screen with a low resolution (<480px) in width (or very low dpi).
3. Scroll down to the comment section and look at the comment submit button (or in German "Kommentar veröffentlichen")

The bug can also be reproduced in a Chrome like browser by going to the developer options, using the "toggle device toolbar" and viewing the website with a width of less than 480px.

**Expected behaviour**
Button should wrap from
"Kommentar veröffentlichen"

to something like:
"Kommentar
veröffentlichen"

Or the German translation should be shorter.

Or a different solution that does not cause an overflow.

**Screenshots**

**Button Overflow on a Phone with a small screen/ low dpi/ low resolution**
![](https://i.imgur.com/AVLkZdk.png)

**Button Overflow with Chrome device toolbar:**
![](https://i.imgur.com/xbd38Mw.png)

**Host Info**
 - OS: Some proxmox kubernetes node running a docker container (or something like that)
 - Wiki.js version: 2.5.170
 - Database engine: postgres 13.1

**Additional context**

As a **workaround** I inject following JavaScript code which replaces "Kommentar veröffentlichen" with "Erstellen". So in English that would mean "Publish Comment" would just become the shorter "Create". **This is obviously not ideal** and I just did it as a quick fix and the page now works on most devices. I am sure that changing the locals would have been a better approach, but I don't know how to do that. I guess the locals de.json are stored somewhere in the docker image and if I would change them, the next time I restart the Wiki.js server they would be gone anyway because in our setup only the DB has a persistent volume.

```js
<script>
function replaceGermanLocals() {
   var x = document.getElementsByClassName("text-none");
   var i;
   for (i = 0; i < x.length; i++) {
      x[i].innerText = x[i].innerText.replace(/Kommentar veröffentlichen/g, 'Erstellen');
   }
};
window.onload = function() {
  replaceGermanLocals();
};
</script>
```

I also want to note that **the bug still occurs with this "quick and dirty fix"**. It just happens at an even lower resolution. While testing with the Chrome Browser: Before "the fix" the Overflow would start happening at 480px, now at about 350px (in width).

Here is an additional screenshot showing that the overflow is now not that bad but still happens on this device:
![](https://i.imgur.com/FRq0GBg.png)

**Comment by mittkkoo** [comment]
This workaround worked fine for me. I used to insert the script in the script section of some pages. Then after rendering the script appeared on the end of the HTML body.

But then I had to insert another script via Adiminstration / Theme / Head HTML injection. I inserted [this script](https://github.com/requarks/wiki/discussions/4580#discussioncomment-1533379)  and after that the dubstition for "Kommentar veröffentlichen" didn't work any more on any site. I tried to insert the script in Body HTML injection (i.e. for all pages). The insertion itself worked and script appeared again at the end of each page's HTML body, but it doesn't do its job and "Kommentar veröffentlichen" remains unchanged.

**Comment by itssme** [comment]
I think the problem is that in my workaround I explicitly set the `window.onload` function. As the other script you linked to also overwrites the `window.onload` function it overwrites the set function from the first script. So either you explicitly call `replaceGermanLocals()` from inside the other `window.onload` function, or you rewrite both scripts to use an event listener like:

```js
window.addEventListener("load",function() {
    replaceGermanLocals():
},false);

window.addEventListener("load",function(){
    # stuff from you other script
},false);
```

I guess that should work, however I can't test it out myself, as we are not using wiki.js anymore.

**Comment by mittkkoo** [comment]
I changed the translation of "Post comment" from "Kommentar veröffentlichen" to "OK" and added a comment with a link to this issue here. The key is `common:comments.postComment`. I found out that everything else, even "Erstellen" is too long and makes a mess in certain layouts.
